perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java

```diff
@@ -137,6 +138,27 @@ public void deleteLastFacilityRemovesConsentHub() throws Exception {
 		assertThatExceptionOfType(ConsentHubAlreadyRemovedException.class).isThrownBy(() -> perun.getConsentsManagerBl().deleteConsentHub(sess, hub));
 	}
 
+	@Test
+	public void removeLastFacilityRemovesConsentHub() throws Exception {
+		System.out.println(CLASS_NAME + "removeLastFacilityRemovesConsentHub");
+		Facility facility = setUpFacility();
+
+		ConsentHub hub = consentsManagerEntry.getConsentHubByFacility(sess, facility.getId());
+
+		perun.getConsentsManagerBl().removeFacility(sess, hub, facility);
+		assertThatExceptionOfType(ConsentHubNotExistsException.class).isThrownBy(() -> perun.getConsentsManagerBl().checkConsentHubExists(sess, hub));
+	}
+
+	@Test
+	public void addFacilityToConsentHubAgain() throws Exception {
```

Comment:
I would probably include another test which covers normal behaviour of addFacility (This test **would** cover it if createFacility() used addFacility() )